resilience4j-bulkhead/src/main/java/io/github/resilience4j/bulkhead/adaptive/AdaptiveBulkhead.java

```
@@ -0,0 +1,575 @@
+	 */
+	static <T> Supplier<T> decorateSupplier(AdaptiveBulkhead bulkhead, Supplier<T> supplier) {
+		return () -> {
+			long start = 0;
+			boolean isFailed=false;
+			bulkhead.acquirePermission();
+			try {
+				start = Instant.now().toEpochMilli();
+				return supplier.get();
+			} catch (Exception e) {
+				isFailed=true;
+				bulkhead.onError( start,TimeUnit.MILLISECONDS, e);
+				throw e;
+			} finally {
+				if (start != 0 && !isFailed) {
+					bulkhead.onSuccess(start, TimeUnit.MILLISECONDS);
+				}
+			}
+		};
+	}
+
+	/**
+	 * Returns a consumer which is decorated by a bulkhead.
+	 *
+	 * @param bulkhead the bulkhead
+	 * @param consumer the original consumer
+	 * @param <T>      the type of the input to the consumer
+	 * @return a consumer which is decorated by a Bulkhead.
+	 */
+	static <T> Consumer<T> decorateConsumer(AdaptiveBulkhead bulkhead, Consumer<T> consumer) {
+		return t -> {
+			long start = 0;
+			boolean isFailed=false;
+			bulkhead.acquirePermission();
+			try {
+				start = Instant.now().toEpochMilli();
+				consumer.accept(t);
+			} catch (Exception e) {
+				isFailed=true;
+				bulkhead.onError( start,TimeUnit.MILLISECONDS, e);
+				throw e;
+			} finally {
+				if (start != 0 && !isFailed) {
+					bulkhead.onSuccess(start, TimeUnit.MILLISECONDS);
+				}
+			}
+		};
+	}
+
+	/**
+	 * Returns a consumer which is decorated by a bulkhead.
+	 *
+	 * @param bulkhead the bulkhead
+	 * @param consumer the original consumer
+	 * @param <T>      the type of the input to the consumer
+	 * @return a consumer which is decorated by a Bulkhead.
+	 */
+	static <T> CheckedConsumer<T> decorateCheckedConsumer(AdaptiveBulkhead bulkhead, CheckedConsumer<T> consumer) {
+		return t -> {
+			long start = 0;
+			boolean isFailed=false;
+			bulkhead.acquirePermission();
+			try {
+				start = Instant.now().toEpochMilli();
+				consumer.accept(t);
+			} catch (Exception e) {
+				isFailed=true;
+				bulkhead.onError( start,TimeUnit.MILLISECONDS, e);
```

Shouldn't it be the total duration instead of start?
long durationInNanos = System.nanoTime() - start;

---

Inside OnError we do that calculation as if it is an ignored error we do not care about it